### PR TITLE
added kotlin demo

### PR DESCRIPTION
**Pull request by twatzl**
Hi,

So I couldn't find any good demo on how to use pf4j with gradle kotlin DSL and I spent days finding out how to translate the buildscripts, especially since the gradle demo also uses some outdated/deprecated stuff.

So I thought I would share the knowledge and wanted to create this full stack Kotlin demo. 

Main points:
- all components are written in Kotlin: app, api, plugins
- using gradle.kts buildscripts for the whole project
- basically an enhanced copy of demo_gradle

Regards,
Tobi

### CODE REVIEW

**Comment by decebals** [comment]
@twatzl Thanks for your contribution. 

Personally, I don't know/use Kotlin so it's hard for me to maintain this new module. I propose you create a new repository (`pf4j-kotlin-demo`) and I will add a link to this demo in documentation, in section `Kotlin`. If you want and you think that is beneficial, you can transfer this repo to `pf4j` organisation for a more visibility. 
In the end I wish the same thing for the existed `demo_gradle` (to create a new home/repo `pf4j-gradle-demo` for him).

Thanks for understating!

**Comment by twatzl** [comment]
That makes total sense for me. I will try to extract it to a separate repo soon and will post a link to the repo here.

**Comment by twatzl** [comment]
Created a repo for the kotlin demo here. I can transfer ownership of the repo and/or add you as admin whatever you would prefer.

https://github.com/twatzl/pf4j-kotlin-demo

**Comment by decebals** [comment]
@twatzl I have updated the [documentation](https://pf4j.org/doc/kotlin.html) to include a link to your repository. Thanks for your contribution!
